exe/Main.hs

```diff
@@ -250,181 +267,184 @@ loadSession dir = do
+          (df, targets) <- evalGhcEnv hscEnv $
+              setOptions opts (hsc_dflags hscEnv)
+          let deps = componentDependencies opts ++ maybeToList hieYaml
+          dep_info <- getDependencyInfo deps
+          -- Now lookup to see whether we are combining with an existing HscEnv
+          -- or making a new one. The lookup returns the HscEnv and a list of
+          -- information about other components loaded into the HscEnv
+          -- (unitId, DynFlag, Targets)
+          modifyVar hscEnvs $ \m -> do
+              -- Just deps if there's already an HscEnv
+              -- Nothing is it's the first time we are making an HscEnv
+              let oldDeps = Map.lookup hieYaml m
+              let -- Add the raw information about this component to the list
+                  -- We will modify the unitId and DynFlags used for
+                  -- compilation but these are the true source of
+                  -- information.
+                  new_deps = RawComponentInfo (thisInstalledUnitId df) df targets cfp opts dep_info
+                                : maybe [] snd oldDeps
+                  -- Get all the unit-ids for things in this component
+                  inplace = map rawComponentUnitId new_deps
+
+              new_deps' <- forM new_deps $ \RawComponentInfo{..} -> do
+                  -- Remove all inplace dependencies from package flags for
+                  -- components in this HscEnv
+                  let (df2, uids) = removeInplacePackages inplace rawComponentDynFlags
+                  let prefix = show rawComponentUnitId
+                  -- See Note [Avoiding bad interface files]
+                  processed_df <- setCacheDir logger prefix (sort $ map show uids) opts df2
+                  -- The final component information, mostly the same but the DynFlags don't
+                  -- contain any packages which are also loaded
+                  -- into the same component.
+                  pure $ ComponentInfo rawComponentUnitId
+                                       processed_df
+                                       uids
+                                       rawComponentTargets
+                                       rawComponentFP
+                                       rawComponentCOptions
+                                       rawComponentDependencyInfo
+              -- Make a new HscEnv, we have to recompile everything from
+              -- scratch again (for now)
+              -- It's important to keep the same NameCache though for reasons
+              -- that I do not fully understand
+              logInfo logger (T.pack ("Making new HscEnv" ++ show inplace))
+              hscEnv <- emptyHscEnv ideNc libDir
+              newHscEnv <-
+                -- Add the options for the current component to the HscEnv
+                evalGhcEnv hscEnv $ do
+                  _ <- setSessionDynFlags df
+                  checkSession logger ghcLibCheck
```

Comment:
I believe that the `ghcLibCheck` is no longer necessary, as the installation check should be accurate enough now that we are comparing the ghc installation that was used to compile *ghcide* with the one that is being used to run it.